packages/devtools_app/lib/src/inspector/inspector_tree_controller.dart

```diff
@@ -211,27 +223,26 @@ class InspectorTreeController extends Object
     if (cachedRows[index]?.isSelected == true) {
       _cachedSelectedRow = cachedRows[index];
     }
-    return cachedRows[index];
+    final cachedRow = cachedRows[index];
+    cachedRow?.isSearchMatch =
+        _searchableCachedRows.safeGet(index)?.isSearchMatch ?? false;
+    return cachedRow;
   }
 
   double getRowOffset(int index) {
     return (getCachedRow(index)?.depth ?? 0) * columnWidth;
   }
 
   List<InspectorTreeRow> getPathFromSelectedRowToRoot() {
-    final selectedItem = _cachedSelectedRow ??
-        cachedRows.firstWhere(
-          (element) => element.isSelected,
-          orElse: () => null,
-        );
+    final selectedItem = _cachedSelectedRow;
     if (selectedItem == null) return [];
 
     final pathToRoot = <InspectorTreeRow>[selectedItem];
     InspectorTreeNode nextParentNode = selectedItem.node.parent;
     while (nextParentNode != null) {
-      final rowItem = cachedRows.firstWhere(
-        (element) => element.node == nextParentNode,
-        orElse: null,
+      final rowItem = _searchableCachedRows.firstWhere(
```

Comment:
have you tried this with a very large app? The worst case scenario of this is that you are trying to get the path from the deepest leaf node to the root. Where N is the depth of the leaf and M is the total number of nodes in the tree, this could be NxM, which could be very slow for a large app.

seems like it would be more performant to either A) make rows aware of their parent rows so that `row.parent` would yield the next parent, or B) to give Node's access to their associated row, so that `row.node.parent.row` would yield the next parent.

Reply:
Turns out it's enough to have an instance of `InspectorTreeNode` for the selected row to find a way to the root. Such instance is available from an instance of `InspectorTreeRow` and `_cachedSelectedRow` in particular. This method now uses parent of the selected `row.node` to find a path to the root. Should be more performant now as it doesn't iterate through the whole tree to find the right item on every iteration of the `while` loop.